packages/amplication-build-manager/src/code-generator/code-generator-splitter.service.ts

```diff
@@ -26,31 +43,93 @@ export class CodeGeneratorSplitterService {
       const serverDSGResourceData: DSGResourceData = cloneDeep(dsgResourceData);
       serverDSGResourceData.resourceInfo.settings.adminUISettings.generateAdminUI =
         false;
-      jobs.push([EnumDomainType.Server, serverDSGResourceData]);
+      const jobBuildId: JobBuildId<BuildId> = `${buildId}-${EnumDomainName.Server}`;
+      await this.setJobStatus(jobBuildId, EnumJobStatus.InProgress);
+      jobs.push([jobBuildId, serverDSGResourceData]);
     }
 
     if (generateAdminUI) {
       const adminUiDSGResourceData: DSGResourceData =
         cloneDeep(dsgResourceData);
       adminUiDSGResourceData.resourceInfo.settings.serverSettings.generateServer =
         false;
-      jobs.push([EnumDomainType.AdminUI, adminUiDSGResourceData]);
+      const jobBuildId: JobBuildId<BuildId> = `${buildId}-${EnumDomainName.AdminUI}`;
+      await this.setJobStatus(jobBuildId, EnumJobStatus.InProgress);
+      jobs.push([jobBuildId, adminUiDSGResourceData]);
     }
 
     return jobs;
   }
 
+  // accumulate the status of the jobs and return the status of the whole build
+  async getBuildStatus(key: BuildId): Promise<EnumJobStatus> {
+    const buildValue = await this.redisService.get<RedisValue>(key);
+    const jobsStatus = Object.values(buildValue);
+
+    const allSucceeded = jobsStatus.every(
+      (status) => status === EnumJobStatus.Success
+    );
+
+    if (allSucceeded) return EnumJobStatus.Success;
+
+    const atLeaseOneFailed = jobsStatus.some(
+      (status) => status === EnumJobStatus.Failure
+    );
+
+    if (atLeaseOneFailed) return EnumJobStatus.Failure;
+
+    const atLeastOneInProgress = jobsStatus.some(
+      (status) => status === EnumJobStatus.InProgress
+    );
+
+    if (atLeastOneInProgress) return EnumJobStatus.InProgress;
+  }
+
+  async getJobStatus(jobBuildId: JobBuildId<BuildId>): Promise<EnumJobStatus> {
+    const key = this.extractBuildId(jobBuildId);
+    const value = await this.redisService.get<RedisValue>(key);
+    this.logger.debug("Job status", { value });
+    return value[jobBuildId];
+  }
+
+  /**
+   * Set / update the status of a job
+   * First it gets the current value of the key (buildId)
+   * Then it updates the value by accessing the jobBuildId property
+   * If the key yet doesn't exist, currentVal will return null, but it won't throw an error because spreading null is ok (obj = {...null} = {})
+   * @param jobBuildId buildId with domain name suffix (server / admin-ui)
+   * @param status EnumJobStatus (in-progress / success / failure)
+   */
+  async setJobStatus(
+    jobBuildId: JobBuildId<BuildId>,
+    status: EnumJobStatus
+  ): Promise<void> {
+    const key = this.extractBuildId(jobBuildId);
+    const currentVal = await this.redisService.get<RedisValue>(key);
+    const newVal = {
+      ...currentVal,
+      [jobBuildId]: status,
+    };
+    this.logger.debug("Updating job status...", { newVal });
+    await this.redisService.set<RedisValue>(key, newVal);
```

Comment:
We could improve performance and avoid race conditions by leveraging Redis Hash map to update part of the value stored.

Reply:
We can open a new issue to improve this.

Reply:
https://github.com/amplication/amplication/issues/7418